admin/controllers/helpdesk.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const moment = require('moment-timezone')
+
+const checkWindowV2Service = require('../services/check-window-v2.service')
+const config = require('../config')
+const pupilRegisterService = require('../services/pupil-register.service')
+const schoolHomeFeatureEligibilityPresenter = require('../helpers/school-home-feature-eligibility-presenter')
+const schoolImpersonationService = require('../services/school-impersonation.service')
+const schoolService = require('../services/school.service')
+const resultPageAvailabilityService = require('../services/results-page-availability.service')
+const ValidationError = require('../lib/validation-error')
+
+const controller = {}
+
+/**
+ * Renders the helpdesk school impersonation page
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @param {object} error
+ * @returns {Promise.<void>}
+ */
+controller.getSchoolImpersonation = async (req, res, next, error = null) => {
+  res.locals.pageTitle = 'MTC Helpdesk School Impersonation'
+  schoolImpersonationService.removeImpersonation(req.user)
+  try {
+    res.render('helpdesk/school-impersonation', {
+      formData: req.body,
+      err: error || new ValidationError(),
+      breadcrumbs: req.breadcrumbs()
+    })
+  } catch (error) {
+    next(error)
+  }
+}
+
+/**
+ * Submit helpdesk school impersonation
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @returns {Promise.<void>}
+ */
+controller.postSchoolImpersonation = async (req, res, next) => {
+  const { dfeNumber } = req.body
+  try {
+    const outcome = await schoolImpersonationService.validateCreateImpersonation(req.user, dfeNumber)
+    if (outcome.hasError && outcome.hasError()) {
+      return controller.getSchoolImpersonation(req, res, next, outcome)
+    }
+    res.redirect('home')
+  } catch (error) {
+    next(error)
+  }
+  return next()
+}
+
+/**
+ * Renders the helpdesk (role) landing page
+ * @param {object} req
+ * @param {object} res
+ * @param {function} next
+ * @returns {Promise.<void>}
+ */
+controller.getHelpdeskHome = async (req, res, next) => {
```

Comment:
is this meant to be a helpdesk version of the school home page?  if so, i would expect us to simply use the existing one.

Reply:
@GuyHarwood there are going to be additional features appearing on this home page.

Reply:
ok, so why don't we just have an optional partial for the PIN history link that is only rendered when its helpdesk?  that feels easier than duplicating all of the ongoing changes that we will make to the school homepage...

Reply:
@GuyHarwood resolving this as it is now using the school home page